File: frontend/src/Dto/profile/DeleteAccountDto.ts
class DeleteAccountDto {
    public deleteAccount: boolean;
    public deactivateAccount: boolean;

    public constructor(deleteAccountRequest:boolean) {
        this.deleteAccount = false;
        this.deactivateAccount = false;
        deleteAccountRequest ? this.deleteAccount = true : this.deactivateAccount = true;
    }
}

export default DeleteAccountDto;

File: backend/src/main/java/com/svelteup/app/backend/profile/dtos/DeleteAccountDto.java
package com.svelteup.app.backend.profile.dtos;

import io.swagger.annotations.ApiModel;
import lombok.NoArgsConstructor;

@ApiModel("DeleteAccountDto is used to carry delete request information from the user.")
public class DeleteAccountDto
{
    public Boolean deleteAccount;
    public Boolean deactivateAccount;

    public DeleteAccountDto()
    {
        this.deleteAccount = false;
        this.deactivateAccount = false;
    }
}
